content/en/tracing/trace_collection/automatic_instrumentation/single-step-apm/compatibility.md

```diff
@@ -0,0 +1,195 @@
+| 3.11           | <i class="icon-check-bold"></i> |       |
+| 3.10           | <i class="icon-check-bold"></i> |       |
+| 3.9            | <i class="icon-check-bold"></i> |       |
+| 3.8            | <i class="icon-check-bold"></i> |       |
+| 3.7            | <i class="icon-check-bold"></i> |       |
+
+
+<div class="alert alert-info">Single Step Instrumentation disables automatically when custom instrumentation is detected.</div>
+
+{{< /programming-lang >}}
+
+{{< programming-lang lang="ruby" >}}
+
+<div class="alert alert-warning">Using Single Step Instrumentation with Ruby applications is in Preview.</div>
+
+### Supported Ruby versions
+
+| Ruby Version | Support | Notes |
+|--------------|---------|-------|
+| 3.2          | Preview |       |
+| 3.1          | Preview |       |
+| 3.0          | Preview |       |
+| 2.7          | Preview |       |
+
+### Troubleshooting
+
+When uninstalling Single Step Instrumentation from a Ruby application, follow these steps to prevent errors:
+
+1. **Before uninstalling**: Make a backup of your `Gemfile` and `Gemfile.lock`.
+2. **After uninstalling**:
+   - Restore your original `Gemfile` and `Gemfile.lock`, or
+   - Run `bundle install` to rebuild your dependencies.
+
+### Additional information
+
+- Requires Linux distributions using glibc 2.27 or newer
+- Not compatible with Alpine Linux or other musl-based distributions
+
+<div class="alert alert-info">Single Step Instrumentation disables automatically when custom instrumentation is detected.</div>
+
+{{< /programming-lang >}}
+
+{{< programming-lang lang="nodejs" >}}
+
+### Supported Node.js versions
+
+| Node.js Version    | Support                         | Notes                                         |
+|--------------------|---------------------------------|-----------------------------------------------|
+| Latest LTS release | <i class="icon-check-bold"></i> | Instrumenting ESM modules is not supported.   |
+
+<div class="alert alert-info">Single Step Instrumentation disables automatically when custom instrumentation is detected.</div>
+
+{{< /programming-lang >}}
+
+{{< programming-lang lang="dotnet" >}}
+
+### Supported .NET runtimes
+
+| .NET Version     | Support                         | Notes                                |
+|------------------|---------------------------------|--------------------------------------|
+| .NET 8           | <i class="icon-check-bold"></i> |                                      |
+| .NET 7           | <i class="icon-check-bold"></i> |                                      |
+| .NET 6           | <i class="icon-check-bold"></i> | Versions below 6.0.13 not supported. |
+| .NET 5           | <i class="icon-check-bold"></i> |                                      |
+| .NET Core 3.1    | <i class="icon-check-bold"></i> |                                      |
+| .NET Core 3.0    |                                 |                                      |
```

Comment:
I'm not sure how to interpret rows that neither have the check mark nor "not supported" - which are they? Are they in a weird third state?

Reply:
Great note. I don't even think we need to mention that the preview versions are explicitly unsupported. I removed that row, so it should be more clear now.